Libraries/PeriphDrivers/Include/MAX32657/i3c.h

```diff
@@ -0,0 +1,621 @@
+    ///< support for SDA signal
+    MXC_I3C_HIGH_KEEPER_EXT_SCL_SDA = MXC_S_I3C_CONT_CTRL0_HKEEP_EXT_SCL_SDA,
+    ///< External high-keeper support for SCL and SDA signals
+} mxc_i3c_high_keeper_t;
+
+/**
+ * @brief   The list of receive FIFO trigger levels.
+ */
+typedef enum {
+    MXC_I3C_RX_TH_NOT_EMPTY = MXC_V_I3C_CONT_FIFOCTRL_RX_THD_LVL_NOT_EMPTY, ///<
+    MXC_I3C_RX_TH_QUARTER_FULL = MXC_V_I3C_CONT_FIFOCTRL_RX_THD_LVL_QUARTER_FULL, ///<
+    MXC_I3C_RX_TH_HALF_FULL = MXC_V_I3C_CONT_FIFOCTRL_RX_THD_LVL_HALF_FULL, ///<
+    MXC_I3C_RX_TH_3_4_FULL = MXC_V_I3C_CONT_FIFOCTRL_RX_THD_LVL_3_QUARTER_FULL, ///<
+} mxc_i3c_rx_threshold_t;
+
+/**
+ * @brief   The list of transmit FIFO trigger levels.
+ */
+typedef enum {
+    MXC_I3C_TX_TH_EMPTY = MXC_V_I3C_CONT_FIFOCTRL_TX_THD_LVL_EMPTY, ///<
+    MXC_I3C_TX_TH_QUARTER_FULL = MXC_V_I3C_CONT_FIFOCTRL_TX_THD_LVL_QUARTER_FULL, ///<
+    MXC_I3C_TX_TH_HALF_FULL = MXC_V_I3C_CONT_FIFOCTRL_TX_THD_LVL_HALF_FULL, ///<
+    MXC_I3C_TX_TH_ALMOST_FULL = MXC_V_I3C_CONT_FIFOCTRL_TX_THD_LVL_ALMOST_FULL, ///<
+} mxc_i3c_tx_threshold_t;
+
+/**
+ * @brief IBI types.
+ *
+ */
+typedef enum {
+    MXC_I3C_IBI_TYPE_NONE = MXC_V_I3C_CONT_STATUS_IBITYPE_NONE, ///<
+    MXC_I3C_IBI_TYPE_IBI = MXC_V_I3C_CONT_STATUS_IBITYPE_IBI, ///<
+    MXC_I3C_IBI_TYPE_CONTROLLER_REQ = MXC_V_I3C_CONT_STATUS_IBITYPE_CONT_REQ, ///<
+    MXC_I3C_IBI_TYPE_HOTJOIN_REQ = MXC_V_I3C_CONT_STATUS_IBITYPE_HOTJOIN_REQ, ///<
+} mxc_i3c_ibi_type_t;
+
+/* CCC request flags */
+#define MXC_I3C_CCC_HAS_DEFINING_BYTE (1U << 0)
+#define MXC_I3C_CCC_HAS_SUB_COMMAND (1U << 1)
+#define MXC_I3C_CCC_HAS_DATA (1U << 2)
+
+/**
+ * @brief Common Command Code request structure.
+ *
+ * Broadcast and direct commands are distinguished by MSB, i.e. if MSB being set
+ * implies a direct CCC while MSB being 0 implies a broadcast CCC.
+ *
+ */
+struct _i3c_ccc_req_t {
+    uint8_t ccc; ///< CCC command to send.
+    uint8_t target_addr : 7; ///< Target address if CCC is a direct command. Ignored
+    ///< if CCC is a broadcast command.
+    uint8_t read_write : 1; ///< Set to 1 for read commands and set to 0 for write
+    ///< commands. Ignored for broadcast CCCs.
+    uint8_t flags; ///< See request flags above.
+    uint8_t def_byte; ///< Optional defining byte. Defined by CCC.
+    uint8_t subCmd; ///< Optional sub-command. Defined by CCC. Only used by direct
+    ///< CCCs.
+    unsigned char *tx_buf; ///< Optional data bytes to send.
+    uint8_t tx_len; ///< Length of optional data.
+    unsigned char *rx_buf; ///< Optional data bytes to read.
+    uint8_t rx_len; ///< Length of optional data.
+};
+
+/**
+ * @brief Private SDR request structure.
+ *
+ * SDR read and write to an I3C target.
+ *
+ */
+struct _i3c_req_t {
+    uint8_t target_addr : 7; ///< Target address.
+    uint8_t is_i2c : 1; ///< If this is a legacy I2C transfer.
+    uint8_t stop; ///< Send a STOP after the transaction.
```

Comment:
Use `bool` for any logical field members (`is_i2c`, `stop`, etc.)

Reply:
Done.